src/Features/Core/Portable/ExternalAccess/UnitTesting/API/UnitTestingSearchHelpers.cs

```diff
@@ -0,0 +1,174 @@
+
+            if (client != null)
+            {
+                var locations = await client.TryInvokeAsync<IRemoteUnitTestingSearchService, ImmutableArray<UnitTestingSourceLocation>>(
+                    project,
+                    (service, solutionChecksum, cancellationToken) => service.GetSourceLocationsAsync(solutionChecksum, project.Id, query, cancellationToken),
+                    cancellationToken).ConfigureAwait(false);
+                if (!locations.HasValue)
+                    return ImmutableArray<UnitTestingDocumentSpan>.Empty;
+
+                using var _ = ArrayBuilder<UnitTestingDocumentSpan>.GetInstance(out var result);
+                foreach (var location in locations.Value)
+                    result.AddIfNotNull(await location.TryRehydrateAsync(project.Solution, cancellationToken).ConfigureAwait(false));
+
+                return result.ToImmutable();
+            }
+
+            return await GetSourceLocationsInProcessAsync(project, query, cancellationToken).ConfigureAwait(false);
+        }
+
+        private static (string containerName, string symbolName, int symbolArity) BuildQueryInfo(UnitTestingSearchQuery query)
+        {
+            if (query.MethodName == null)
+            {
+                // if we don't have a method name, then the fully qualified type name needs to be broken into two parts:
+                // 1. the name of the type symbol we're looking for (the last name portion of the qualified name)
+                // 2. the container of the type symbol we're looking for (all but the last name portion).
+                var fullyQualifiedTypeName = query.FullyQualifiedTypeName;
+                var lastPlus = fullyQualifiedTypeName.LastIndexOf('+');
+                var lastDot = fullyQualifiedTypeName.LastIndexOf('.');
+
+                var (container, type) =
+                    lastPlus >= 0 ? (fullyQualifiedTypeName[..lastPlus], fullyQualifiedTypeName[(lastPlus + 1)..]) :
+                    lastDot >= 0 ? (fullyQualifiedTypeName[..lastDot], fullyQualifiedTypeName[(lastDot + 1)..]) :
+                    ("", fullyQualifiedTypeName);
+
+                GetNameAndArity(type, out type, out var typeArity);
+
+                return (ConvertTypeName(container), type, typeArity);
+            }
+            else
+            {
+                // If we have a method name, then that's the name we'll search in the index for. The fully qualified
+                // type name is what we'll use to check the container of any methods we find.
+                return (ConvertTypeName(query.FullyQualifiedTypeName), query.MethodName, query.MethodArity);
+            }
+        }
+
+        /// <summary>
+        /// Converts from a metadata-name into the internal simple dotted name we store in our index as the container
+        /// for a symbol.  In the future, we could consider storing the fully-qualified-metadata-name in our index as
```

Comment:
indeed, would be nice to not have to do string manipulation to do this.  

I assume we don't expect metdata name syntax to change much?

Reply:
metadata-name-syntax def won't change.  it's a standard for the runtime.  sicne we only have to manipulate hte string once here, and then seasrch the index after doing that it's not a big deal IMO :)